### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by dryaz**
Implement setting to use fill and stroke option for outer control
Add feature to avoid mask fields (use plain numbers)
Implement error state of the view (errorColor stroke)

### CODE REVIEW

File: passcodeview/src/main/java/online/devliving/passcodeview/PasscodeView.java [hunk 1]
```diff
@@ -221,7 +244,7 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             public void afterTextChanged(Editable s) {
                 int length = s.length();
                 updateChilViewSelectionStates(length, mEditText.hasFocus());
-
+                mPasscodeEntryListener.onPasscodeDigitEntered();
```
Comment (iamMehedi) [inline passcodeview/src/main/java/online/devliving/passcodeview/PasscodeView.java]: @dryaz shouldn't we check if `mPasscodeEntryListener` is null here?

File: README.md [hunk 2]
```diff
@@ -38,13 +38,20 @@ compile 'online.devliving:passcodeview:1.0.3'
 - `setText(CharSequence text)` - Set Passcode programmatically
 - `clearText()` - Clear Passcode
 - `getText()` - get entered Passcode
+- `swetError(boolean value)` - set the error state of view
```
Comment (iamMehedi) [inline README.md]: I guess that's a typo, should be `setError`?

**Review by iamMehedi — CHANGES_REQUESTED** [review]
Thanks for the contribution. please take a look at the comments.

File: README.md (hunk 2 above)
Comment (dryaz) [reply README.md]: You're right, yes, thx, will update after my vacation in ~1 week

File: passcodeview/src/main/java/online/devliving/passcodeview/PasscodeView.java (hunk 1 above)
Comment (dryaz) [reply passcodeview/src/main/java/online/devliving/passcodeview/PasscodeView.java]: You're right, yes, thx, will update after my vacation in ~1 week

**Review by iamMehedi — APPROVED** [review]
looks good!

**Comment by dryaz** [comment]
> looks good!

Feel free to merge it :)

**Comment by dryaz** [comment]
@iamMehedi do you have plans to merge this PR or should I close it?

**Pull request closed without merging**